django/core/signing.py

```diff
@@ -183,6 +159,44 @@ def unsign(self, signed_value):
             return value
         raise BadSignature('Signature "%s" does not match' % sig)
 
+    def sign_object(self, obj, serializer=JSONSerializer, compress=False):
+        """
+        Return URL-safe, hmac signed base64 compressed JSON string.
+
+        If compress is True (not the default), check if compressing using zlib can
+        save some space. Prepend a '.' to signify compression. This is included
+        in the signature, to protect against zip bombs.
+
+        The serializer is expected to return a bytestring.
+        """
+        data = serializer().dumps(obj)
+        # Flag for if it's been compressed or not
+        is_compressed = False
+
+        if compress:
+            # Avoid zlib dependency unless compress is being used
+            compressed = zlib.compress(data)
+            if len(compressed) < (len(data) - 1):
+                data = compressed
+                is_compressed = True
+        base64d = b64_encode(data).decode()
+        if is_compressed:
+            base64d = '.' + base64d
+        return self.sign(base64d)
+
+    def unsign_object(self, signed_obj, serializer=JSONSerializer, **kwargs):
```

Comment:
Both methods should be documented in ["Verifying timestamped values"](https://docs.djangoproject.com/en/3.1/topics/signing/#django.core.signing.TimestampSigner) with new examples in ["Using the low-level API"](https://docs.djangoproject.com/en/3.1/topics/signing/#using-the-low-level-api). Also, release notes and some adjustments in ["Protecting complex data structures"](https://docs.djangoproject.com/en/3.1/topics/signing/#protecting-complex-data-structures) are required.

Reply:
Sure thing, thanks for the links. I will plan to adjust the docs tomorrow.

Reply:
@felixxm will this be for 3.1.5 or 3.2?

Reply:
My mistake - I can see in the trac ticket this bug fix is for 3.1

Reply:
It doesn't qualify for a backport, also we don't add new features after the feature freeze, so please target this to 3.2.

Reply:
I've updated the PR with the change in the release notes.
 
One thing I noticed in another part of the documentation, is that the `sign_object` and `unsign_object` docs don't automatically provide links to the source code. I guess that this might be because they are methods of the base `Signer` class, which `TimestampSigner`inherits, but they are currently doc'd in the `TimestampSigner` . Is there some trick to getting them to link to the source code from the base class?